pkg/tool/cmdexecutor.go

```diff
@@ -0,0 +1,28 @@
+package tool
+
+import (
+	"strings"
+	"text/template"
+
+	"github.com/helm/chart-testing/v3/pkg/exec"
+)
+
+type CmdTemplateExecutor struct {
+	exec exec.ProcessExecutor
+}
+
+func NewCmdTemplateExecutor(exec exec.ProcessExecutor) CmdTemplateExecutor {
+	return CmdTemplateExecutor{
+		exec: exec,
+	}
+}
+
+func (t CmdTemplateExecutor) RunCommand(cmdTemplate string, data interface{}) error {
+	var template = template.Must(template.New("command").Parse(cmdTemplate))
+	var b strings.Builder
+	if err := template.Execute(&b, data); err != nil {
+		return err
+	}
+	renderedCommand := b.String()
+	return t.exec.RunProcess("sh", "-c", renderedCommand)
```

Comment:
This means given user needs to have a shell installed, which may be a problem on other platforms (especially on Windows).
I believe it should be a raw RunProcess, in that case someone can explicitly define how to run commands or other tool to invoke commands (for example in another container, but that would be container in container in some cases....).

Reply:
I would be fine with executing it without a shell @unguiculus what's your opinion?

Reply:
The windows argument only holds true if you execute chart testing locally. If you use the built container it's always linux and there using a shell could be preferable.

Reply:
One more thing, the 'sh' may run different shell than other people would expect - I refer bash vs dash in here, especially under os x.

Reply:
Most freedom would be to simply leave it out. I mean users could add `sh -c` in their additional command. I just want to get feedback from @unguiculus as he suggested to add it in the first place.

Reply:
From https://github.com/helm/chart-testing/pull/283#discussion_r499725440

> What if an argument or a path contains spaces? You should use a shell to run the process.

These could then be wrapped in quotes, yes? Aren't we already properly wrapping quotes?

Reply:
You need to somehow separate command from args. If a user configures "sh -c 'mytool'" and you pass that as command, it wouldn't work.

Maybe have a look at https://github.com/mattn/go-shellwords, which would allow us to parse the commandline and run it using an OS-specific shell.

Reply:
Yep 👍  [func Command(name string, arg ...string) *Cmd](https://golang.org/pkg/os/exec/#Command) needs a command `name` string, then variadic `args`.

Did a quick playground test with go-shellwords and seems to work as expected: https://play.golang.org/p/H9Q9MO_0e2R Nice find 😄

Reply:
@unguiculus Thanks for the hint. command and args are now spitted with   https://github.com/mattn/go-shellwords